### PR TITLE
honor projected name for parameter

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/3858

- honor @projectedName for operation parameter
- add one testcases for special-words

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/utils.ts [hunk 1]
```diff
@@ -52,3 +55,14 @@ export function getModelName(context: SdkContext, m: Model): string {
 
     return name;
 }
+
+export function getProjectedNameForCsharp(
```
Comment (ArcturusZhang) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/utils.ts:59]: based on the purpose of this function, I think we should name it as `getNameForCSharp`

**Review by ArcturusZhang — APPROVED** [review]

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts [hunk 2]
```diff
@@ -259,7 +259,7 @@ export function loadOperation(
                     : InputOperationParameterKind.Client
                 : InputOperationParameterKind.Method;
         return {
-            Name: param.name,
+            Name: getProjectedNameForCsharp(sdkContext, param) ?? param.name,
```
Comment (lirenhe) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts:262]: As this is inside c# emitter, do we need to include csharp in function name?

Comment (lirenhe) [reply src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts:262]: I am wondering whether we could also put ' ?? param.name' inside function body and just call it getTypeName()

Comment (chunyu3) [reply src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts:262]: Different Type (Operation, Model, ModelProperty) may have different name logic, some may have friendlyName, or some may get the Name to combine the template args. And it is in a order, such as if projectedName is undefined, and try to get FriendlyName, if also undefined, then it origin name.
So we cannot combine all those in `getTypeName` and we need to just handle projectedName and return undefined if it is not defined.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**